### PR TITLE
fix(browser-history): decode fragment

### PR DESCRIPTION
**Pull request by bigopon**
Without decoding fragment, URL with special character will trigger `_loadUrl` twice.

@EisenbergEffect @davismj 

fixes /aurelia/router#630

### CODE REVIEW

File: src/browser-history.ts [hunk 1]
```diff
@@ -39,7 +37,7 @@ export class BrowserHistory extends History {
     super();
 
     this._isActive = false;
-    this._checkUrlCallback = this._checkUrl.bind(this);
+    this._checkUrl = this._checkUrl.bind(this);
```
Comment (EisenbergEffect) [inline src/browser-history.ts:40]: Any reason why this change was made?

Comment (bigopon) [reply src/browser-history.ts:40]: The old way is less bundle size friendly

Comment (EisenbergEffect) [reply src/browser-history.ts:40]: Doesn't this change the behavior though, since the same property is representing both a bound and unbound function?

Comment (bigopon) [reply src/browser-history.ts:40]: Yes, it changes instance behavior. Anyone who wants prototype fn can still get it via `BrowserHistory.prototype._checkUrl`.
